src/vanilla.ts

```diff
@@ -109,6 +119,7 @@ export default function create<TState extends State>(
 
   const destroy: Destroy = () => listeners.clear()
   const api = { setState, getState, subscribe, destroy }
-  state = createState(setState, getState, api)
+  // SetState<TState> includes CustomSetState so it should be ok
```

Comment:
Can we do something like this?
```diff
diff --git a/src/vanilla.ts b/src/vanilla.ts
index ed33bc2..38d1fd7 100644
--- a/src/vanilla.ts
+++ b/src/vanilla.ts
@@ -62,7 +62,7 @@ export type StateCreator<T extends State, CustomSetState = SetState<T>> = (
 
 export default function create<
   TState extends State,
-  CustomSetState = SetState<TState>
+  CustomSetState extends SetState<TState> = SetState<TState>
 >(createState: StateCreator<TState, CustomSetState>): StoreApi<TState> {
   let state: TState
   const listeners: Set<StateListener<TState>> = new Set()
@@ -119,7 +119,6 @@ export default function create<
 
   const destroy: Destroy = () => listeners.clear()
   const api = { setState, getState, subscribe, destroy }
-  // SetState<TState> includes CustomSetState so it should be ok
-  state = createState(setState as any as CustomSetState, getState, api)
+  state = createState(setState as CustomSetState, getState, api)
   return api
 }
```

Reply:
Ahh,I should have figure it out before using `any`... Thank you for good catch!

Reply:
Hm, like a test I added, I'd like to apply a `CustomSetState` that set a different type of state.
In a example, `set` cannot update `readA/B` because the second argument is `SetState<Pick<ExampleState, 'a' | 'b'>>`. However, `Pick<ExampleState, 'a' | 'b'>` couldn't satisfies `TState`...

```typescript
  CustomSetState extends SetState<any> = SetState<TState>
```

is going to fix `setState as any as CustomSetState` but test passes. I'll send a patch for your check.

Reply:
Yeah, that's something I was concerned. Maybe this can't be helped.